### PR TITLE
bt-345: Get badges and hard skills from api

### LINKED ISSUE #345: Get badges and hard skill from API
After - https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/issues/290 

We should connect the client to API

**Acceptance criteria:**

- Talent should be able to choose hard skills and badges from the list of real data

### PR DESCRIPTION
**Pull request by vladjsp**
closes https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/issues/345

https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/118971905/5fa313bc-f70d-4fbb-9cd0-5d8907bcff4b

### CODE REVIEW

File: frontend/src/bundles/common-data/types/bsa-badges-response-dto.type.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+type BsaBadgesResponseDto = {
+    items: {
+        id: string;
+        type: string;
+        name: string;
+        maxScore: number;
+    }[];
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common-data/types/bsa-badges-response-dto.type.ts]: Create type for this items

File: frontend/src/bundles/common-data/types/hard-skills-response-dto.type.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+type HardSkillsResponseDto = {
+    items: {
+        id: string;
+        name: string;
+    }[];
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common-data/types/hard-skills-response-dto.type.ts]: The same

File: frontend/src/bundles/common-data/store/actions.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+});
+
+const getBsaBadgesData = createAsyncThunk<
+    BsaBadgesResponseDto,
+    undefined,
+    AsyncThunkConfig
+>(`${sliceName}${ApiPath.BSA_BADGES}`, async (_, { extra }) => {
+    const { commonDataApi } = extra;
+
+    return await commonDataApi.getAllBsaBadges();
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common-data/store/actions.ts]: Don't use return await

File: frontend/src/bundles/common-data/store/actions.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import { name as sliceName } from './slice.js';
+
+const getHardSkillsData = createAsyncThunk<
+    HardSkillsResponseDto,
+    undefined,
+    AsyncThunkConfig
+>(`${sliceName}${ApiPath.HARD_SKILLS}`, async (_, { extra }) => {
+    const { commonDataApi } = extra;
+
+    return await commonDataApi.getAllHardSkills();
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common-data/store/actions.ts]: The same

File: frontend/src/bundles/common-data/common-data-api.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        const response = await this.load(
+            this.getFullEndpoint(ApiPath.HARD_SKILLS, {}),
+            {
+                method: 'GET',
+                contentType: ContentType.JSON,
+                hasAuth: true,
+            },
+        );
+
+        return await response.json<HardSkillsResponseDto>();
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common-data/common-data-api.ts]: The same

File: frontend/src/bundles/common-data/common-data-api.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        const response = await this.load(
+            this.getFullEndpoint(ApiPath.BSA_BADGES, {}),
+            {
+                method: 'GET',
+                contentType: ContentType.JSON,
+                hasAuth: true,
+            },
+        );
+
+        return await response.json<BsaBadgesResponseDto>();
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common-data/common-data-api.ts]: The same

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

**Review by savinaDaria — APPROVED** [review]
Tested, badges and hard-skills are returned.

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
Discussed in Slack

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**